packages/fiori-elements-writer/src/index.ts

```diff
@@ -17,6 +17,33 @@ import { changesPreviewToVersion, escapeFLPText } from './data/templateAttribute
 import { extendManifestJson } from './data/manifestSettings';
 import semVer from 'semver';
 
+export const V2_FE_TYPES_AVAILABLE = '1.108.0';
+/**
+ * Get TypeScript Ignore Glob Pattern.
+ *
+ * @param feApp  to generate the Fiori elements application
+ * @param coercedUI5Version
+ * @returns ignore pattern
+ */
+function getTypeScriptIgnoreGlob<T extends {}>(feApp: FioriElementsApp<T>, coercedUI5Version: semVer.SemVer): string[] {
+    let ignore = [];
+    // isV2FETypesAvailable - Boolean to indicate if V2 Fiori Element types were available in the UI5 version
+    const isV2FETypesAvailable = feApp.ui5?.version ? semVer.gte(coercedUI5Version, V2_FE_TYPES_AVAILABLE) : false;
+    const tsIgnoreGlob = ['**/*.js'];
+    ignore = tsIgnoreGlob;
+    // Add local ui5.d.ts if types are missing in UI5 version for V2 Odata services
+    // OR template is OVP
+    if (feApp.service.version === OdataVersion.v2) {
+        if (isV2FETypesAvailable) {
```

Comment:
Should this check be reversed? Should we ignore `ui5.d.ts` if the types are missing and not the other way around?

Reply:
@devinea clarified to me that the check is correct - we're not copying the types if they're available. Thanks!